superset/sql_lab.py

```diff
@@ -322,13 +362,22 @@ def execute_sql_statements(
         key = str(uuid.uuid4())
         logging.info(f"Storing results in results backend, key: {key}")
         with stats_timing("sqllab.query.results_backend_write", stats_logger):
-            json_payload = json.dumps(
-                payload, default=json_iso_dttm_ser, ignore_nan=True
-            )
+            with stats_timing(
+                "sqllab.query.results_backend_write_serialization", stats_logger
+            ):
+                serialized_payload = _serialize_payload(
+                    payload, results_backend_use_msgpack
+                )
             cache_timeout = database.cache_timeout
             if cache_timeout is None:
                 cache_timeout = config.get("CACHE_DEFAULT_TIMEOUT", 0)
-            results_backend.set(key, zlib_compress(json_payload), cache_timeout)
+
+            compressed = zlib_compress(serialized_payload)
+            logging.debug(
+                f"*** serialized payload size: {getsizeof(serialized_payload)}"
```

Comment:
Since this is an optimization PR, you may want to: 
```python
if logging.getLogger().getEffectiveLevel() == logging.DEBUG:
    payload_size = getsizeof(serialized_payload)
    logging.debug("*** serialized payload size: %s", payload_size)
```
Super nitty: you may want to avoid f-strings in logging messages to avoid serializing the string if the logging level is higher. Meaning if you logging level is `logging.INFO` you won't serialize the msg. But that would only matter for things that get called a lot (not the case here...).